addons/lunch/models/lunch_order.py

```diff
@@ -39,121 +39,85 @@ def _default_previous_order_ids(self):
-                lunch_data[line.id] = {
-                    'line_id': line.id,
-                    'product_id': line.product_id.id,
-                    'product_name': line.product_id.name,
-                    'supplier': line.supplier.name,
-                    'note': line.note,
-                    'price': line.price,
-                    'date': fields.Date.to_string(line.date),
-                    'currency_id': line.currency_id.id,
-                }
-            # sort the old lunch orders by (date, id)
-            lunch_data = OrderedDict(sorted(lunch_data.items(), key=lambda t: (t[1]['date'], t[0]), reverse=True))
-            self.previous_order_widget = json.dumps(lunch_data)
-
-    @api.one
     @api.depends('user_id')
     def _compute_cash_move_balance(self):
+        self.ensure_one()
         domain = [('user_id', '=', self.user_id.id)]
         lunch_cash = self.env['lunch.cashmove'].read_group(domain, ['amount', 'user_id'], ['user_id'])
         if len(lunch_cash):
             self.cash_move_balance = lunch_cash[0]['amount']
         self.balance_visible = (self.user_id == self.env.user) or self.user_has_groups('lunch.group_lunch_manager')
 
-    @api.one
     @api.constrains('date')
     def _check_date(self):
         """
         Prevents the user to create an order in the past
         """
+        self.ensure_one()
         date_order = self.date
         date_today = fields.Date.context_today(self)
         if date_order < date_today:
             raise ValidationError(_('The date of your order is in the past.'))
 
-    @api.one
-    @api.depends('order_line_ids.state')
-    def _compute_order_state(self):
-        """
-        Update the state of lunch.order based on its orderlines. Here is the logic:
-        - if at least one order line is cancelled, the order is set as cancelled
-        - if no line is cancelled but at least one line is not confirmed, the order is set as new
-        - if all lines are confirmed, the order is set as confirmed
-        """
-        if not self.order_line_ids:
-            self.state = 'new'
-        else:
-            isConfirmed = True
-            for orderline in self.order_line_ids:
-                if orderline.state == 'cancelled':
-                    self.state = 'cancelled'
-                    return
-                elif orderline.state == 'confirmed':
-                    continue
-                else:
-                    isConfirmed = False
+    def action_order(self):
+        for order in self:
+            for line in order.order_line_ids:
+                self.env['lunch.cashmove'].sudo().create({
+                    'user_id': line.user_id.id,
+                    'amount': -line.price,
+                    'description': line.product_id.name,
+                    'order_id': line.id,
+                    'state': 'order',
+                    'date': line.date,
+                })
+                line.write({'state': 'ordered'})
```

Comment:
Apply the `write` on the `self.mapped('lines')` to execute only one query ;)
Same on the orders.